packages/README.md

```diff
@@ -0,0 +1,167 @@
+npm version 6.7.8 --workspace=packages/apple
+
+# Increment the version of each package to the next minor version
+npm version minor --workspaces
+
+## Increment the version of a package to a pre-release version of the next minor version
+npm version preminor --workspace=packages/apple
+```
+
+When a subpackage imports another, it should be set to depend on the `*` version of the other subpackage.
+
+### Adding a new subpackage
+
+To set up a new package:
+
+1. Add a `package.json` and `README.md` for that package.
+1. Ensure all source files are located in `src` and all compiled files are built to `lib`.
+1. Add an entry to the `workspaces` section of the root `package.json` so that NPM is aware of your package.
+1. Set up import aliases so that the package is visible from the web app to the following tools:
+    1. Webpack - Add an entry to the `resolve.alias` section of the root `webpack.config.js` pointing to the `src` folder set up above.
+
+    ```js
+    module.exports = {
+        resolve: {
+            alias: {
+                '@mattermost/apple': 'packages/apple/src',
+            },
+        },
+    };
+    ```
+
+    1. TypeScript - In the root `tsconfig.json`, add an entry to the `compilerOptions.paths` section pointing to the `src` folder and an entry to the `references` section pointing to the root of your package which should contain its own `tsconfig.json`.
+
+    Note that the `compilerOptions.paths` entry will differ based on if your package exports just a single module (ie a single `index.js` file) or if it exports multiple submodules.
+
+    ```json
+    {
+        "compilerOptions": {
+            "paths": {
+                "@mattermost/apple": ["packages/apple/src"], // import * as Apple from '@mattermost/apple';
+                "@mattermost/banana/*": ["packages/banana/src/*"], // import Yellow from '@mattermost/banana/yellow';
+            }
+        },
+        "references": [
+            {"path": "./packages/apple"},
+            {"path": "./packages/banana"},
+        ]
+    }
+    ```
+
+    1. Jest - Add an entry to the `jest.moduleNameMapper` section of the root `package.json` for your package. Since that setting supports regexes, you can add these to the existing patterns used by the `client` and `types` packages.
+
+    Similar to TypeScript, this will differ based on if the package exports a single module or multiple modules.
+
+    ```json
+    {
+        "jest": {
+            "moduleNameMapper": {
+                "^@mattermost/(apple|client)$": "<rootDir>/packages/$1/src",
+                "^@mattermost/(banana|types)/(.*)$": "<rootDir>/packages/$1/src/$2",
+            }
+        }
+    }
+    ```
+
+### Publishing a subpackage
+
+The following is the rough process for releasing these packages. They'll require someone with write access on our NPM organization to run them, and they'll likely change over time as we improve this process.
+
+For full releases accompanying new versions of Mattermost:
```

Comment:
This'll probably end up being automated eventually so that you just tag a release and GitHub or GitLab does all this for us